### ISSUE TITLE
Hope to support diff-quality in pre-commit hooks.

### ISSUE THREAD

**Issue by maoxingda** [sqlfluff/sqlfluff#5451]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

**The following command runs successfully from the command line, but when placed in a pre-commit hook, it does not detect violations**

```
conda run -n env-name diff-quality --violations sqlfluff --compare-branch origin/main
```

```yaml
repos:
  - repo: local
    hooks:
      - id: sqlfluff
        name: sqlfluff
        entry: conda run -n env-name diff-quality --violations sqlfluff --compare-branch origin/main
        language: system
        types: [sql]
```

### Dialect

no

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by oleg-savko** [comment]
Also have this trouble.

@maoxingda or someone, do you found workaround maybe?

**Comment by maoxingda** [comment]
Not yet, I'm in trouble.

**Comment by peterbud** [comment]
Hi @maoxingda  - thanks for opening this issue.

If I'm not mistaken, `diff-quality` is intended for CI repo-level diff checks, while `pre-commit` is intended for per-file hooks. They overlap conceptually, but they are not shaped the same way operationally.

If you want to keep using diff-quality in pre-commit, you might try fixing the issue by disabling filename passing with adding option
`pass_filenames: false` to the hook.

However you might be better by running diff-quality in CI instead of pre-commit as described in the [docs](https://docs.sqlfluff.com/en/stable/production/diff_quality.html#adding-diff-quality-to-your-builds)

So the reason you see this difference is because pre-commit changes the invocation semantics.

Let me know if you have more questions.

**Comment by peterbud** [comment]
Closing due to lack of response
